pkg/ee/metering/cronjob.go

```diff
@@ -38,136 +40,33 @@ import (
 )
 
 // cronJobCreator returns the func to create/update the metering report cronjob.
-func cronJobCreator(seedName, reportName string, mrc *kubermaticv1.MeteringReportConfiguration, getRegistry registry.WithOverwriteFunc) reconciling.NamedCronJobCreatorGetter {
+func cronJobCreator(reportName string, mrc *kubermaticv1.MeteringReportConfiguration, getRegistry registry.WithOverwriteFunc, namespace string) reconciling.NamedCronJobCreatorGetter {
 	return func() (string, reconciling.CronJobCreator) {
 		return reportName, func(job *batchv1.CronJob) (*batchv1.CronJob, error) {
-			labels := job.GetLabels()
-			if labels == nil {
-				labels = make(map[string]string)
+			var args []string
+			args = append(args, fmt.Sprintf("--prometheus-api=http://%s.%s.svc", prometheus.Name, namespace))
+			args = append(args, fmt.Sprintf("--last-number-of-days=%d", mrc.Interval))
+			args = append(args, fmt.Sprintf("--output-dir=%s", reportName))
+			args = append(args, mrc.Types...)
+
+			if job.Labels == nil {
+				job.Labels = make(map[string]string)
 			}
-			labels[LabelKey] = reportName
-			job.SetLabels(labels)
+			job.Labels[common.NameLabel] = reportName
+			job.Labels[common.ComponentLabel] = meteringName
 
 			job.Spec.Schedule = mrc.Schedule
-			job.Spec.JobTemplate.Spec.Parallelism = pointer.Int32Ptr(1)
-			job.Spec.JobTemplate.Spec.Template.Spec.ServiceAccountName = meteringToolName
```

Comment:
I think this causes an issue for upgraded clusters, the fact `ServiceAccountName` is not explicitly removed causes old `CronJobs` to contain the old value and that `ServiceAccount` is purged as a legacy resource in the mechanism introduced in this PR.
https://github.com/kubermatic/kubermatic/blob/8fd688856e73375558ecb9969e34388158d0aff6/pkg/ee/metering/reconcile.go#L251-L253